Assets/Scripts/BulletManagement/SlipTimeEmitter.cs

```diff
@@ -11,14 +11,100 @@ public class SlipTimeEmitter : EmitterBase, ISlipTimeAdherent
+        // The number of bullet waves to emit. Set to zero for no limit.
+        public int emitCycles = 0;
+
+        // Number of bullets to emit each emit cycle.
+        public int emitBulletCount;
+        
+        // Angle at which the emtiter points
+        public float startAngle = 0f;
+
+        // The rate at which the emitter should spin each update cycle
+        public float spinRate = 0f;
+
+        // The maximum spin rate at which spin will reverse
+        public float spinMax = 1f;
+        
+        // Analogous to angular acceleration, the rate at which spinRate increases
+        public float spinAccel = 0f;
+        
+        // Flag used to control if the spin should reverse upon hitting max speed
+        public bool reverseSpin = false;
+        
+        // Determines if the emitter should aim at the player or not. Note that this will overwrite startAngle.
+        public bool aimed = false;
+
+        void Start()
+        {
+            limited = emitCycles != 0;
+            this.bulletPattern = this.GetComponent<BulletPattern>();
+            if (aimed)
+            {
+                var angleToPlayer = AngleToPlayer();
+                transform.Rotate(Vector3.forward * -angleToPlayer, Space.Self);
+            }
+            else
+            {
+                transform.Rotate(Vector3.forward * startAngle, Space.Self);
+            }
+
+        }
+        
+        protected float AngleToPlayer()
+        {
+            var playerTransform = GameObject.Find("Player").transform;
+            var emitterTransform = transform;
+            Vector3 targetDir = playerTransform.position - emitterTransform.position;
+            return Vector3.SignedAngle(targetDir, emitterTransform.up, Vector3.forward);
+        }
+
+        private void EmitBullets()
+        {
+            // Instantiate the bulletPrefab at emitter position
+            for (int id = 0; id < this.emitBulletCount; ++id)
+            {
+                var currentTransform = this.transform;
+                GameObject bulletGO = Instantiate(bulletPrefab, currentTransform.position, currentTransform.rotation);
+                Bullet bullet = bulletGO.GetComponent<Bullet>();
+                bullet.SetData(this, this.bulletPattern, id);
+                bullet.decayTime = bulletDecayTime;
+            }
+        }
+        
         // Update is called once per frame
         private void Update()
         {
-            timeCounter += Time.deltaTime * bulletSpeedMultiplier * SlipTimeManager.slipTimeCoefficient;
-            if (timeCounter >= emitFrequency)
+            // Destroy the emitter if it has finished all its cycles (when applicable)
+            if(limited && cyclesEmitted >= emitCycles) Destroy(this);
+            if (aimed)
             {
-                EmitBullets();
-                timeCounter = 0f;
+                var angleToPlayer = AngleToPlayer();
+                transform.Rotate(Vector3.forward * -angleToPlayer, Space.Self);
```

Comment:
Would it be better to use `Vector3.RotateTowards()` instead? https://docs.unity3d.com/ScriptReference/Vector3.RotateTowards.html?_ga=2.28954682.727368189.1605214950-1973212993.1604629452

Reply:
I'm looking through the scripting reference, and I'm not clear on what the performance gains would be by using that method. It looks like we'd either have to (a) still compute the angle between the two anyway to pass in as `maxRadiansDelta` or just set that value to `Math.Inf`, which then means there may be cases where we "take the long way around" in our rotation because we aren't taking advantage of sign. We would also need to figure out the implications of the `maxMagnitudeDelta` -- it seems like that's used to implement behavior we don't need for what we're doing. Maybe I'm missing something? We could potentially save a function call here, I guess.

Reply:
Performance might be better because Unity might have optimized the calculations to a lower API call. The function would also would reduce our implementation by a function and a few calculations. 

Wouldn't bounding `maxRadiansDelta` by PI guarantee minimum rotation?

I think `maxMagnitudeDelta` should be set to 0.0f as seen in the example so that magnitude doesn't change.

Reply:
After further discussion, decided to shelve this for a future optimization pass because it wasn't as straightforward to implement as it looked at first blush.